CMakeLists.txt

```diff
@@ -64,13 +64,16 @@ add_subdirectory(sdk/iot/common)
 add_subdirectory(sdk/iot/hub)
 add_subdirectory(sdk/iot/provisioning)
 
-# PAL
+#PAL
 if(AZ_PLATFORM_IMPL STREQUAL "POSIX")
   add_subdirectory(sdk/platform/posix)
+  set(PAL az_posix)
 elseif(AZ_PLATFORM_IMPL STREQUAL "WIN32")
   add_subdirectory(sdk/platform/win32)
+  set(PAL az_win32)
 else()
   add_subdirectory(sdk/platform/noplatform)
+  set(PAL az_noplatform)
```

Comment:
I see you went with this, I like it :)

There is also fourth case, when AZ_PLATFORM_IMPL is USER, then PAL = AZ_USER_PLATFORM_IMPL_NAME.

And as of the naming... We don't use `AZ_` prefix anywhere else, I propose to rename AZ_PLATFORM_IMPL to PLATFORM, and make both AZ_USER_PLATFORM_NAME and PAL to be PLATFORM_LIB (BTW, we try to avoid word "PAL" in the naming, as it is considered MSFT-internal terminology that is not widely known outside of MS; that is what Jeffrey said at one point; plus we call it "platform" everywhere in our codebase).

Reply:
I see.  makes sense. 
But, is it worth it?  since this `${PAL}` is only for internal use in our cmake projects.

I would like to remove the AZ_USER_PLATFORM_IMPL_NAME.  Because our libraries don't really care about this.  We would still build our samples with az_noplatforms, and user will take care of linking his own implementation to his application instead of using any of ours.

The reason why I added this PAL variable is because in latest master we are only building one of the platforms libs. Either posix, win32 or no_platform. In the past we were always building them.
First approach was to have az_noplatform as always building and only conditioning az_posix and az_win32 (similar how we always build az_nohttp but condition az_curl). But I didn't want to touch the fix from iot, and prefer to have the platform explicitly set